docs/contracts/StMatic.md

```diff
@@ -406,233 +307,109 @@ Requirements:
-  uint16 _insuranceFeeBasisPoints,
-  uint16 _operatorsFeeBasisPoints
-)
-```
-
-#### Parameters
-
-| Name                       | Type     | Description                                                                            |
-| -------------------------- | -------- | -------------------------------------------------------------------------------------- |
-| `_treasuryFeeBasisPoints`  | `uint16` | Fee for the treasury. Expressed in basis points, 10000 BP corresponding to 100%.       |
-| `_insuranceFeeBasisPoints` | `uint16` | Fee for the insurance fund. Expressed in basis points, 10000 BP corresponding to 100%. |
-| `_operatorsFeeBasisPoints` | `uint16` | Fee for the node operators. Expressed in basis points, 10000 BP corresponding to 100%. |
-
-### setOracle()
-
-Set authorized oracle contract address to `_oracle`
-
-```sol
-function setOracle(address _oracle)
-```
-
-#### Parameters
+| `_amount` | `uint256` | Amount to withdraw. |
 
-| Name      | Type      | Description                |
-| --------- | --------- | -------------------------- |
-| `_oracle` | `address` | Address of oracle contract |
+---
 
-### setTreasury()
-
-Set treasury contract address to `_treasury`. This contract is used to accumulate the protocol treasury fee
-
-```sol
-function setTreasury(address _treasury)
+### delegate()
+Delegates the MATIC deposited into the StMatic contract equally to the validators 
+```solidity
+function delegate() 
 ```
 
-#### Parameters
-
-| Name        | Type      | Description                                        |
-| ----------- | --------- | -------------------------------------------------- |
-| `_treasury` | `address` | Address of contract which accumulates treasury fee |
-
-### setInsuranceFund()
-
-Set insuranceFund contract address to `_insuranceFund`.
-This contract is used to accumulate the protocol insurance fee
+---
 
-```sol
-function setInsuranceFund(address _insuranceFund)
+### claimTokens()
+Allows users claim their tokens from the validators. This requires that the user has a pending withdrawal request
+```solidity
+function claimTokens(uint256 _tokenId)
 ```
-
 #### Parameters
 
-| Name             | Type      | Description                                         |
-| ---------------- | --------- | --------------------------------------------------- |
-| `_insuranceFund` | `address` | Address of contract which accumulates insurance fee |
-
-### setWithdrawalCredentials()
-
-Set credentials to withdraw ETH on ETH 2.0 side after the phase 2 is launched to `_withdrawalCredentials`
+| Name              | Type     | Description                                                    |
+| ----------------- | -------- | -------------------------------------------------------------- |
+| `_tokenId` | `uint256` | Token ID of a pending withdrawal request. |
+---
 
-```sol
-function setWithdrawalCredentials(bytes32 _withdrawalCredentials)
+### distributeRewards()
+Distributes rewards claimed from validators to their operators
```

Comment:
Distribute rewards accumulated by lido validators. 90% of rewards are buffered and redelegated. 10% is distributed between DAO(2.5%), insurance(2.5%), and active operators(5%).